File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:

        l = 0 
        r = len(nums) - 1

        while l<=r:

            m = (l + r) // 2
            print(nums[m])

            if nums[l] <= nums[r]:
                return nums[l]

            if l + 1 == r:
                return nums[r]
            elif nums[l] > nums[m]:
                r = m 
            elif nums[l] < nums[m]:
                l = m 
    

File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {

        std::vector<int> single = {};
        for(int i=0; i<nums.size(); i++){

            
            bool exists = checkArr(single, nums[i]);

            if(exists){
                return true;
            }
            else{
                single.push_back(nums[i]);
            }

        }

        return false;
    }

    bool checkArr(vector<int> &single, int nums){
        for(int i = 0; i<single.size(); i++){
            if (single[i] == nums){
                return true;
            }
        }
        return false;
    }
};
